### PR TITLE
Kolibri automatic provisioning: Change default settings

### PR DESCRIPTION
**Pull request by manuq**
Add more facility and device settings. Use defaults more proper for
the seamless user management case.

https://phabricator.endlessm.com/T32885

### CODE REVIEW

File: config/defaults.ini [hunk 1]
```diff
@@ -244,6 +244,11 @@ automatic_provision_superuser_name =
 automatic_provision_superuser_password =
 automatic_provision_preset = nonformal
 automatic_provision_landing_page = learn
+automatic_provision_learner_can_edit_username = false
+automatic_provision_learner_can_edit_name = false
+automatic_provision_learner_can_edit_password = false
+automatic_provision_learner_can_sign_up = false
+automatic_provision_allow_guest_access = false
```
Comment (dylanmccall) [inline config/defaults.ini:251]: I'll be happy with this if we can keep these settings as they are upstream (_unset_ if it is possible?). Instead, something like `learner_can_edit_name = false` should be in the configuration for a specific image.

File: hooks/image/61-kolibri-content-install [hunk 2]
```diff
@@ -17,9 +17,15 @@ if [ "${EIB_KOLIBRI_AUTOMATIC_PROVISION}" = "true" ]; then
...
-  "facility_settings": {},
+  "facility_settings": {
+    "learner_can_edit_username": ${EIB_KOLIBRI_AUTOMATIC_PROVISION_LEARNER_CAN_EDIT_USERNAME},
+    "learner_can_edit_name": ${EIB_KOLIBRI_AUTOMATIC_PROVISION_LEARNER_CAN_EDIT_NAME},
+    "learner_can_edit_password": ${EIB_KOLIBRI_AUTOMATIC_PROVISION_LEARNER_CAN_EDIT_PASSWORD},
+    "learner_can_sign_up": ${EIB_KOLIBRI_AUTOMATIC_PROVISION_LEARNER_CAN_SIGN_UP}
+  },
   "device_settings": {
     "landing_page": "${EIB_KOLIBRI_AUTOMATIC_PROVISION_LANDING_PAGE}",
+    "allow_guest_access": ${EIB_KOLIBRI_AUTOMATIC_PROVISION_ALLOW_GUEST_ACCESS},
```
Comment (dylanmccall) [inline hooks/image/61-kolibri-content-install:28]: :+1: I'm glad this works!

**Review by dylanmccall — CHANGES_REQUESTED** [review]

File: config/defaults.ini (hunk 1 above)
Comment (manuq) [reply config/defaults.ini:251]: Yeah good point. It's a bit of a pain to build a JSON from Bash, but you are right.

Comment (manuq) [reply config/defaults.ini:251]: After a second though I tend to disagree, see my comment in phabricator. https://phabricator.endlessm.com/T32885#919180

Comment (manuq) [reply config/defaults.ini:251]: One thing we could do is decuple the device "can guests access" setting from the rest which are valid in all cases.

**Review by dylanmccall — APPROVED** [review]
Sorry, this all got buried in Phabricator. I think I'm happy with this. A couple of the defaults still feel a bit funky to me, but they're just that: defaults. And we have a matching Kolibri release now so it should all work nicely. Let's merge it and see how it goes :)

**Pull request merged**